### PR TITLE
feat: ChatGuard coreplugin

### PR DESCRIPTION
**Pull request by Martinz64**

### CODE REVIEW

File: src/core-plugins/ChatGuard.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    default?: true;
+};
+
+export default class ChatGuard extends Plugin {
+    private getByName(defaultName: string, options?: FilterOptions) {
+        return getModule(m => m?.default?.name === defaultName, options);
+    }
+
+    CHANNEL_IDS = [
+        "811255667469582420", // #offtopic
```
Comment (janisslsm) [inline src/core-plugins/ChatGuard.tsx:20]: why is offtopic included

**Review by janisslsm — CHANGES_REQUESTED** [review]

File: src/core-plugins/ChatGuard.tsx (hunk 1 above)
Comment (rushiiMachine) [reply src/core-plugins/ChatGuard.tsx:20]: its in kt plugin too

File: src/core-plugins/ChatGuard.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                                <ReactNative.View
+                                    style={{
+                                        //"backgroundColor": "#36393f",
+                                        "borderColor": "#202225",
+                                        "borderTopWidth": 0.36363636363636365,
+                                        "padding": 6,
+                                        "paddingVertical": 8,
+                                        "paddingLeft": 4,
+                                        "paddingRight": 4
+                                    }}>
```
Comment (janisslsm) [inline src/core-plugins/ChatGuard.tsx:56]: you will move styles outside of patch

**Review by janisslsm — CHANGES_REQUESTED** [review]

File: src/core-plugins/ChatGuard.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import { getModule, React, ReactNative, Dialog, Forms, Styles} from "../metro";
+const { FormRow } = Forms;
+import { getAssetId } from "../utils";
+type FilterOptions = {
+    exports?: boolean;
+    default?: false;
+} | {
+    exports?: true;
+    default?: true;
+};
```
Comment (janisslsm) [inline src/core-plugins/ChatGuard.tsx:12]:
```suggestion
```

You shall remove

**Review by janisslsm — CHANGES_REQUESTED** [review]

File: src/core-plugins/ChatGuard.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
+import { Plugin } from "../entities";
+// @ts-ignore
```
Comment (rushiiMachine) [inline src/core-plugins/ChatGuard.tsx:2]: what does this suppress

File: src/core-plugins/ChatGuard.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+        const ChatInputGuard = this.getByName("ChatInputGuard");
+        this.patcher.after(ChatInputGuard.default.prototype, "render", (_, component: any) => {
+            const chatInput = component.props.children.find(c => c?.props?.accessibilityLabel != undefined);
+            const channelId = chatInput?.props?.channel?.id;
+            if(this.CHANNEL_IDS.includes(channelId+"")){
+                component.props.children = [
+                    <>
+                        {
+                            // @ts-ignore
```
Comment (rushiiMachine) [inline src/core-plugins/ChatGuard.tsx:56]: what is this supressing

File: src/core-plugins/ChatGuard.tsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+                                        onPress={()=>{
+                                            Dialog.show({
+                                                title: "PLEASE READ",
+                                                body: this.MESSAGE,
+                                                confirmText: "I Understood",
+                                                cancelText: "Cancel",
+                                                isDismissable: true,
+                                                // @ts-ignore
+                                                onConfirm: ()=>{settings.set("AcknowlegedNoSupportChannels", true);}
+                                            });
```
Comment (rushiiMachine) [inline src/core-plugins/ChatGuard.tsx:75]: is there a way to disable the button for like 5s

File: src/core-plugins/ChatGuard.tsx [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+                                        label={this.MESSAGE_SHORT}
+                                        leading={<FormRow.Icon source={getAssetId("ic_warning_24px")} />}
+                                        trailing={FormRow.Arrow}
+
+                                        onPress={()=>{
+                                            Dialog.show({
+                                                title: "PLEASE READ",
+                                                body: this.MESSAGE,
+                                                confirmText: "I Understood",
+                                                cancelText: "Cancel",
```
Comment (rushiiMachine) [inline src/core-plugins/ChatGuard.tsx:71]: there shoudn't be a way to cancel

File: src/core-plugins/ChatGuard.tsx [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+        "811262084968742932", // #core-development
+
+        //'985647046755221575', // testing channel
+    ];
+
+    MESSAGE = "PLEASE READ: This is not a support channel, do not ask for help! This is NOT A SUPPORT CHANNEL. Do NOT ask for help about using or installing a plugin or theme here or you will be muted.";
+    MESSAGE_SHORT = "PLEASE READ: This is not a support channel, do not ask for help!";
+
+    public async start() {
+        // @ts-ignore
```
Comment (rushiiMachine) [inline src/core-plugins/ChatGuard.tsx:33]: what does this suppress

**Review by rushiiMachine — CHANGES_REQUESTED** [review]

File: src/core-plugins/ChatGuard.tsx (hunk 6 above)
Comment (nikosszzz) [reply src/core-plugins/ChatGuard.tsx:63]: afaik no theres no way

File: src/core-plugins/ChatGuard.tsx [hunk 9]
```diff
@@ -0,0 +1,67 @@
+import { Plugin } from "../entities";
+import { getModule, React, ReactNative, Dialog, Forms, Styles, getByName } from "../metro";
```
Comment (rushiiMachine) [inline src/core-plugins/ChatGuard.tsx:2]:
```suggestion
import { React, ReactNative, Dialog, Forms, Styles, getByName } from "../metro";
```

**Review by rushiiMachine — APPROVED** [review]

**Pull request merged**